thinkcompany/code-challenges
# SEPTA Rail Fare Calculator Challenge (By Lisa MacCarrigan)

**Objective**
-
The challenge was to create an interactive widget for calculating SEPTA Regional Rail fare prices.

**Live Widget:**  [https://septa-fare-calculator.firebaseapp.com/](url) 

**Comments:**
- 

- _Browser Support_: Tested Google Chrome, Firefox, and Safari for iOS (desktop & iPhone 8)
- _Libraries & Frameworks_: I opted for a more Vanilla approach since I have used libraries and frameworks extensively over the past year
- _Standards_: I believe my HTML markup is relatively semantic and the code is performant.
- _Time_: Exceeded the 60 minute limit to deliver a higher-quality product because my portfolio is outdated. I spent roughly 120 minutes on this project.

**Things I Would Revisit:** 🤔
-
- _Browser Support_: Test IE 8+ and Chrome for Android
- _Standards_: Must improve accessibility
- Refactor and improve the `calculateFare` function because it modifies DOM elements and functions should be discrete in purpose. Also, address the console error regarding property `length` of undefined
- Investigate and resolve bugs related to when a user selects the 'anytime' ride option
- Add input validation wherever beneficial (Note: I'm looking at **you**, numeric input field)
- Fix `index.html` indentation. Looks normal in Atom, but on GitHub it is incorrect.
- Replace use of HTML Details (`<details>`) and Disclosure Summary (`<summary>`) elements because they are not fully supported
- Update JSON data because the SEPTA fares are outdated
